src/main/java/software/amazon/event/kafkaconnector/EventBridgeResult.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package software.amazon.event.kafkaconnector;
+
+import static software.amazon.event.kafkaconnector.EventBridgeResult.ErrorType.PANIC;
+import static software.amazon.event.kafkaconnector.EventBridgeResult.ErrorType.REPORT_ONLY;
+import static software.amazon.event.kafkaconnector.EventBridgeResult.ErrorType.RETRY;
+
+import org.apache.kafka.connect.sink.SinkRecord;
+import software.amazon.event.kafkaconnector.util.MappedSinkRecord;
+
+public class EventBridgeResult<T> {
+
+  public enum ErrorType {
+    REPORT_ONLY,
+    RETRY,
+    PANIC
+  }
+
+  public static class Error {
+
+    private final ErrorType type;
+    private final String message;
+    private final Throwable cause;
+
+    private Error(final ErrorType type, final String message, final Throwable cause) {
+      this.message = message;
+      this.cause = cause;
+      this.type = type;
+    }
+
+    public static Error reportOnly(final String message, final Throwable cause) {
+      return new Error(REPORT_ONLY, message, cause);
+    }
+
+    public static Error reportOnly(final String message) {
+      return new Error(REPORT_ONLY, message, null);
+    }
+
+    public static Error retry(final Throwable cause) {
+      return new Error(RETRY, "", cause);
```

Comment:
question: could there be a case where we need a message?

Reply:
Currently not. If we need it in future we can add an overloaded method.